### PR TITLE
Added --output-dir-use-symlinks for tune download with default as auto

### LINKED ISSUE #813: tune download has no option to set local_dir_use_symlinks
For `tune download`, It would be helpful to have `local_dir_use_symlinks` option for the user to choose `auto`,`True` or `False`.
Currently it is set to `auto` which stores the files in .cache directory and uses symlinks for big files.

Ref: https://huggingface.co/docs/huggingface_hub/en/guides/download

### PR DESCRIPTION
**Pull request by apthagowda97**
#### Context
- Fixes #813 
- Added --output-dir-use-symlinks  option to `tune download` with default as `auto`.

#### Changelog
- 

#### Test plan
-

### CODE REVIEW

File: torchtune/_cli/download.py [hunk 1]
```diff
@@ -67,6 +68,18 @@ def _add_arguments(self) -> None:
             default="./model",
             help="Directory in which to save the model.",
         )
+        self._parser.add_argument(
+            "--output-dir-use-symlinks",
+            type=str,
+            required=False,
+            default="False",
```
Comment (joecummings) [inline torchtune/_cli/download.py:75]: Can we default this to 'auto' to mimic the behavior from before?

File: torchtune/_cli/download.py [hunk 2]
```diff
@@ -86,11 +99,28 @@ def _add_arguments(self) -> None:
     def _download_cmd(self, args: argparse.Namespace) -> None:
         """Downloads a model from the Hugging Face Hub."""
         # Download the tokenizer and PyTorch model files
+
+        # Raise if local_dir_use_symlinks is invalid
+        output_dir_use_symlinks: Union[Literal["auto"], bool]
+        use_symlinks_lowercase = args.output_dir_use_symlinks.lower()
```
Comment (joecummings) [inline torchtune/_cli/download.py:105]: `argparse` can actually handle all of this logic within the `add_argument` method.

e.g.

```
self._parser.add_argument(
    ...,
    choices = ["true", "false", "auto"],
)
```

**Review by joecummings — COMMENTED** [review]
Left a couple comments, but overall looks good! Can you also please sign the CLA?

File: torchtune/_cli/download.py (hunk 1 above)
Comment (apthagowda97) [reply torchtune/_cli/download.py:75]: Sure

**Review by joecummings — APPROVED** [review]
Can you install and run the linter?

File: torchtune/_cli/download.py (hunk 2 above)
Comment (apthagowda97) [reply torchtune/_cli/download.py:105]: This wont work because coz.. `true` and `false` should be of type Boolean.

Comment (apthagowda97) [reply torchtune/_cli/download.py:105]: But here the they convert to Boolean, I just used there logic  https://github.com/huggingface/huggingface_hub/blob/54e4f76594d2d8338ddccab566a9750cfe4da1bd/src/huggingface_hub/commands/download.py#L137

Comment (joecummings) [reply torchtune/_cli/download.py:105]: Ahh I missed that in the code, thanks for pointing it out. It's so gross, but seems like the best way to go forward. 

Thanks for all your work on this @apthagowda97, I'll go ahead and merge!

**Pull request merged**